src/app/articles/articles.component.ts

```diff
@@ -8,17 +8,55 @@ import { ArticlesService } from '../articles.service';
 })
 export class ArticlesComponent implements OnInit {
 
-  articles:any = []
+  articles:any = [];
   tags: any = [];
+  selectedTag: any;
+  user: any = false;
   constructor(private service: ArticlesService) { }
 
   ngOnInit(): void {
-
+    this.loadTags();
   }
+
   loadArticles() {
     this.service.getArticles().subscribe(response => this.articles = response.articles);
   }
+
   loadTags() {
     this.service.getTags().subscribe(response => this.tags = response.tags);
   }
+
+  loadArticlesByTag(tag){
+    this.selectedTag = tag;
+    this.service.getArticlesByTag(this.selectedTag).subscribe(response => this.articles = response.articles);
+  }
+
+  classActive(tag)
+  {
+    if(this.selectedTag == tag){
```

Comment:
en este caso no hace falta el if poria ser directamente
return this.selectedTag == tag;

tampoco esta muy bien usar el == mejor el ===  fijate en javasript the good parts el libro